com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/ZigBeeNetworkManager.java

```diff
@@ -514,7 +517,9 @@ public ZigBeeStatus startup(boolean reinitialize) {
      * Shuts down ZigBee manager components.
      */
     public void shutdown() {
-        executorService.shutdownNow();
+        if (executorService != null) {
```

Comment:
See above, the networkmanger instance should "cleanly" survive a `shutdown` and `initialize`. Therefore if we close the the pool in shutdown and access it later, this will result in exceptions.

Reply:
Ok, so this is a change as it's not something that was considered in the past. Previously I always assumed that once shutdown, we would not restart.
I would need to think if the overall system can cope with this - eg do all the extensions restart properly or does something get stuck?

Reply:
Using https://github.com/openhab/org.openhab.binding.zigbee/pull/335 with this PR merged leads to a working zigbee network after I un-plug and re-plug the dongle.

But I could only test it so far with a OSRAM lightbulb and a centralite smart plug.